ngraph/test/backend/floor_mod.in.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <algorithm>
+#include <cinttypes>
+#include <cmath>
+#include <cstdlib>
+#include <random>
+#include <string>
+
+// clang-format off
+#ifdef ${BACKEND_NAME}_FLOAT_TOLERANCE_BITS
+#define DEFAULT_FLOAT_TOLERANCE_BITS ${BACKEND_NAME}_FLOAT_TOLERANCE_BITS
+#endif
+
+#ifdef ${BACKEND_NAME}_DOUBLE_TOLERANCE_BITS
+#define DEFAULT_DOUBLE_TOLERANCE_BITS ${BACKEND_NAME}_DOUBLE_TOLERANCE_BITS
+#endif
+// clang-format on
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/engine/test_engines.hpp"
+#include "util/test_case.hpp"
+#include "util/test_control.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static string s_manifest = "${MANIFEST}";
+using TestEngine = test::ENGINE_CLASS_NAME(${BACKEND_NAME});
+
+NGRAPH_TEST(${BACKEND_NAME}, floor_mod_int32)
+{
+    Shape shape{4};
+
+    auto A = make_shared<op::Parameter>(element::i32, shape);
+    auto B = make_shared<op::Parameter>(element::i32, shape);
+    auto f = make_shared<Function>(make_shared<op::FloorMod>(A, B), ParameterVector{A, B});
+
+    std::vector<int32_t> a{7, -7, 7, -7};
+    std::vector<int32_t> b{3, 3, -3, -3};
+
+    auto test_case = test::TestCase<ngraph::test::INTERPRETER_Engine>(f);
```

Comment:
You missed change `ngraph::test::INTERPRETER_Engine` to `test::ENGINE_CLASS_NAME(${BACKEND_NAME})` that you have under `using TestEngine`. So floor_mod backend tests marked as `IE_CPU` probably still relay on the `INTERPRETER_Engine`.